libfreerdp/core/gateway/rdg.c

```diff
@@ -409,16 +406,18 @@ static BOOL rdg_write_websocket(rdpRdg* rdg, wStream* sPacket)
 	// mask as much as possible with 32bit access
 	for (streamPos = 0; streamPos + 4 <= s; streamPos += 4)
 	{
-		uint32_t masked = *((uint32_t*)(inStream + streamPos)) ^ maskingKey;
-		Stream_Write_UINT32(sWS, masked);
+		uint32_t data;
+		Stream_Read_UINT32(sPacket, data);
```

Comment:
it must be "same endian" as where the masking key is transmitted a few lines above:
Stream_Write_UINT32(sWS, maskingKey);
since maskingKey itself is random it does not really matter for the actual "number" itself

If you want I can change both to UNINT32_BE

Reply:
if it is by intention it is ok, I was just curious (did not want to dig too deep) if network byte order is required or not

Reply:
side note: to me it looks more readably in little endian because a few lines down there is this:

BYTE* partialMask = ((BYTE*)&maskingKey) + (streamPos % 4);

this is essentially litte endian.